manim/cli/render/render_options.py

```diff
@@ -2,40 +2,103 @@
 
 import logging
 import re
+from typing import TYPE_CHECKING
 
 from cloup import Choice, option, option_group
 
 from manim.constants import QUALITIES, RendererType
 
+if TYPE_CHECKING:
+    from click import Context, Option
+
 __all__ = ["render_options"]
 
 logger = logging.getLogger("manim")
 
 
-def validate_scene_range(ctx, param, value):
+def validate_scene_range(
+    ctx: Context, param: Option, value: str | None
+) -> tuple[int] | tuple[int, int] | None:
+    """If the ``value`` string is given, extract from it the scene range, which
+    should be in any of these formats: 'start', 'start;end', 'start,end' or
+    'start-end'. Otherwise, return ``None``.
+
+    Parameters
+    ----------
+    ctx
+        The Click context.
+    param
+        A Click option.
+    value
+        The optional string which will be parsed.
+
+    Returns
+    -------
+    tuple[int] | tuple[int, int] | None
+        If ``value`` is ``None``, the return value is ``None``. Otherwise, it's
+        the scene range, given by a tuple which may contain a single value
+        ``start`` or two values ``start`` and ``end``.
+
+    Raises
+    ------
+    ValueError
+        If ``value`` has an invalid format.
+    """
+    if value is None:
+        return None
+
     try:
         start = int(value)
         return (start,)
     except Exception:
         pass
 
-    if value:
-        try:
-            start, end = map(int, re.split(r"[;,\-]", value))
-            return start, end
-        except Exception:
-            logger.error("Couldn't determine a range for -n option.")
-            exit()
+    try:
+        start, end = map(int, re.split(r"[;,\-]", value))
+    except Exception:
+        logger.error("Couldn't determine a range for -n option.")
+        exit()
```

Comment:
We should probably use `sys.exit` here, `exit()` is only intended for use in the REPL.

Reply:
Right! I just made that change.